components/script/dom/serviceworkercontainer.rs

```diff
@@ -25,72 +26,83 @@ use std::rc::Rc;
 pub struct ServiceWorkerContainer {
     eventtarget: EventTarget,
     controller: MutNullableHeap<JS<ServiceWorker>>,
+    client: MutNullableHeap<JS<Client>>,
 }
 
 impl ServiceWorkerContainer {
     fn new_inherited() -> ServiceWorkerContainer {
         ServiceWorkerContainer {
             eventtarget: EventTarget::new_inherited(),
             controller: Default::default(),
+            client: Default::default(),
         }
     }
 
+    #[allow(unrooted_must_root)]
     pub fn new(global: &GlobalScope) -> Root<ServiceWorkerContainer> {
-        reflect_dom_object(box ServiceWorkerContainer::new_inherited(), global, Wrap)
+        let container = ServiceWorkerContainer::new_inherited();
+        let client = Client::new(&global.as_window());
+        container.client.set(Some(&*client));
+        reflect_dom_object(box container, global, Wrap)
```

Comment:
I propose we create the client first, and pass it as an argument to new_inherited instead. That should let us use `JS<Client>` instead.